Documentation/README_Solver.md

```diff
@@ -36,5 +38,19 @@ When *UpdateLinkedTransform* is true, the solver will calculate position & orien
 ## Expectations for extending or adding to the solver system ##
 To create a new solver script, you will want to extend from the abstract base class, *Solver*. This will ensure your new solver will tie into the state tracking and execution from both Solver and SolverHandler. The power with solvers comes from them being modular in nature, so as you extend Solver, keep this in mind. Its better to have many small solver scripts vs. one big one.
 
+## Hand constraint solvers ##
+
+The **HandConstraint** behavior provides a solver that constrains the tracked object to a region safe for hand constrained content (such as hand UI, menus, etc). Safe regions are considered areas that don't intersect with the hand. A derived class of **HandConstraint** called **HandConstraintPalmUp** is also included to demonstrate a common behavior of activating the solver tracked object when the palm is facing the user. For example use of this behavior please see the **HandBasedMenuExample** scene.
+
+Please see the tool tips available for each **HandConstraint** property for additional documentation. A few properties are defined in more detail below.
+
+<img src="../Documentation/Images/Solver/MRTK_Solver_HandConstraintPalmUp.PNG" width="450">
+
+* **Safe Zone**: The safe zone specifies where on the hand to constrain content. It is recommended that content be placed on the Ulnar Side to avoid overlap with the hand and improved interaction quality. Safe zones are calculated by taking the hands orientation projected into a plane orthogonal to the camera's view and raycasting against a bounding box around the hands.
+
+<img src="../Documentation/Images/Solver/MRTK_Solver_HandConstraintSafeZones.PNG" width="450">
+
+* **Activation Events**: Currently the **HandConstraint** triggers four events, OnHandActivate, OnHandDecativate, OnFirstHandDetected, and OnLastHandDetected. The OnHandActivate events triggers when a hand satisfies the IsHandActive method and OnHandDecativate triggers when the IsHandActive method is no longer satisfied. OnFirstHandDetected occurs when state changes from no hands in view, to the first hand in view. And, on OnLastHandDetected occurs when state changes from at least one hand in view, to no hands in view. These events can be used in many different combinations to create unique **HandConstraint** behaviors, please see the **HandBasedMenuExample** scene for examples of these behaviors.
+
```

Comment:
Nit: Do you think we should mention the requirement of HandsBounds? or just leave it be since ideally developers don't even need to reference it or mess with it

Reply:
I could go either way, they kind of exist behind the scenes aren't dependent on the Solver system. So maybe they could be documented elsewhere?